external-crates/move/crates/move-vm-runtime/src/tracing2/tracer.rs

```diff
@@ -559,16 +591,42 @@ impl VMTracer<'_> {
         Some(())
     }
 
-    fn close_initial_frame_(&mut self, return_values: &[Value], remaining_gas: u64) -> Option<()> {
+    fn close_initial_native_frame_(
+        &mut self,
+        return_values: &[Value],
+        remaining_gas: u64,
+    ) -> Option<()> {
         let current_frame_return_tys = self.current_frame()?.return_types.clone();
         let return_values: Vec<_> = return_values
             .iter()
             .zip(current_frame_return_tys.into_iter())
             .map(|(value, tag_with_layout_info_opt)| {
                 let (layout, ref_type) = tag_with_layout_info_opt.layout;
                 let move_value = value.as_annotated_move_value_for_tracing_only(&layout?)?;
-                assert!(ref_type.is_none());
-                Some(TraceValue::RuntimeValue { value: move_value })
+                match ref_type {
```

Comment:
Looks very similar to the code in `open_initial_frame`. Would it be possible to refactor?